src/components/modals/ManageGroupsAndTypesModal.svelte

```diff
@@ -0,0 +1,679 @@
+      return derivationGroup.event_types.includes(selectedExternalEventType.name);
+    } else {
+      return false;
+    }
+  });
+
+  async function deleteDerivationGroup(derivationGroup: DerivationGroup) {
+    // Makes sure all associated sources are deleted before this. List of sources already contained in DerivationGroup type.
+    await showDeleteDerivationGroupModal(derivationGroup, user);
+  }
+
+  async function deleteExternalSourceType(sourceType: ExternalSourceType) {
+    let associatedDerivationGroups = $derivationGroups.filter(
+      derivationGroup => derivationGroup.source_type_name === sourceType.name,
+    );
+
+    // makes sure all associated derivation groups are deleted before this
+    await showDeleteExternalEventSourceTypeModal(
+      sourceType,
+      'External Source Type',
+      associatedDerivationGroups.map(derivationGroup => derivationGroup.name),
+      user,
+    );
+  }
+
+  async function deleteExternalEventType(eventType: ExternalEventType) {
+    const associatedDerivationGroups: DerivationGroup[] = getAssociatedDerivationGroupsByEventType(eventType.name);
+    const associatedExternalSourceNames: string[] = associatedDerivationGroups.flatMap(derivationGroup =>
+      Array.from(derivationGroup.sources.keys()),
+    );
+
+    // makes sure all associated sources (and therefore events, as orphans are not possible) are deleted before this
+    // NOTE: does not update in derivation_group_comp after removing a EE type; derivation_group_comp defaults to 0 event types after its last external source removed,
+    //        as it has no awareness of external source type or paired events (as the latter don't even exist).
+    await showDeleteExternalEventSourceTypeModal(eventType, 'External Event Type', associatedExternalSourceNames, user);
+  }
+
+  function getAssociatedExternalSourcesBySourceType(sourceType: string | undefined) {
+    if (sourceType === undefined) {
+      return [];
+    }
+    let associatedSources = $externalSources.filter(source => source.source_type_name === sourceType);
+    return associatedSources;
+  }
+
+  function getAssociatedDerivationGroupsBySourceTypeName(sourceTypeName: string | undefined) {
+    if (sourceTypeName === undefined) {
+      return [];
+    }
+
+    let associatedDerivationGroups = $derivationGroups.filter(
+      derivationGroup => derivationGroup.source_type_name === sourceTypeName,
+    );
+
+    return associatedDerivationGroups;
+  }
+
+  function getAssociatedDerivationGroupsByEventType(eventType: string | undefined) {
+    if (eventType === undefined) {
+      return [];
+    }
+    const associatedDerivationGroups: DerivationGroup[] = $derivationGroups.filter(derivationGroup =>
+      derivationGroup.event_types.includes(eventType),
+    );
+
+    return associatedDerivationGroups;
+  }
+
+  function viewDerivationGroup(viewedDerivationGroup: DerivationGroup) {
+    const derivationGroup = $derivationGroups.find(
```

Comment:
Yep! If `derivationGroup` is `undefined` it falls to the `else` which is used to reset the modal to 'remove'/'hide' the detail column to the right. I believe this is the expected outcome as if no derivation group is found, we've got nothing to expand for detail

Reply:
But it looks like there's no check for if `derivationGroup` is undefined. If `selectedDerivationGroup` is undefined, then I think it falls into the `if` regardless if `deviationGroup` is defined or not and could potentially set the the `modalColumnSize`

Reply:
I think the reasoning for not having a check here is because `viewedDerivationGroup` necessarily comes from the list of derivation groups, so it _has_ to match one of the items in `$derivationGroups`. This find seems like an extra check to get the root data in case anything defers, though that should not even be possible, hence the lack of a check. @JosephVolosin we can either remove the find and just use `viewedDerivationGroup`, or go with the safest though potentially unnecessary option, which is to add a check that `derivationGroup` is not null in the `if` and `else` clauses, and do nothing if `derivationGroup` itself is null (which should never be possible).

Reply:
Ah, I see the problem now - believe this is now resolved after checking that `derivationGroup !== undefined` in the `if` statement!